### DISCUSSION TITLE
Does load_collection load non-vector fields into RAM?

### DISCUSSION THREAD

**Discussion by Wiley-Z** [milvus-io/milvus#34879] [Q&A and General discussion] [answered]
I've inserted large VARCHAR or ARRAY in collections. I found that RAM expenditure while load_collection was obviously bigger than index files, like 700m to 5GB, 450GB to over 700GB (not enough RAM to know exactly). Does load_collection load every field into RAM, not just index? If it does, is there anything I can do to avoid loading no-vector fields?

**Comment by xiaofan-luan** [comment, marked as answer]
you can mmap to local disk to save memory.

if this field don't need to be filtered or retrieved, maybe it's a good idea to not write these data into milvus

**Reply by Wiley-Z** [reply]
call `collection.set_properties({'mmap.enabled': True})` and `collection.alter_index(index_name=index_name, extra_params={"mmap.enabled": False})`, than index is loaded in RAM and other fields are on disk. Is that correct?

**Reply by xiaofan-luan** [reply]
yep
